libcudacxx/include/cuda/std/__tuple_dir/ignore.h

```diff
@@ -0,0 +1,42 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2023 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _LIBCUDACXX___TUPLE_IGNORE_H
+#define _LIBCUDACXX___TUPLE_IGNORE_H
+
+#include <cuda/std/detail/__config>
+
+#if defined(_CCCL_IMPLICIT_SYSTEM_HEADER_GCC)
+#  pragma GCC system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_CLANG)
+#  pragma clang system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_MSVC)
+#  pragma system_header
+#endif // no system header
+
+_LIBCUDACXX_BEGIN_NAMESPACE_STD
+
+template <class _Up>
+struct __ignore_t
```

Comment:
Q: I don't understand why we need `__ignore_t` to be a template. It's an ordinary struct in the paper. I assume this a workaround to affect the emited symbol? @miscco ?

Reply:
I don't know either. Only libc++ implemented `std::ignore` like that. Both STL and stdc++ and now also newer libc++ implement it without the template. But I rather didn't touch it.

Reply:
Alright, let's make it an ordinary struct.

Reply:
I believe that was a historical artifact form our libc++ fork